common/djangoapps/dark_lang/tests.py

```diff
@@ -42,33 +43,70 @@ def setUp(self):
             enabled=True
         ).save()
 
-    def process_request(self, language_session_key=UNSET, accept=UNSET, preview_lang=UNSET, clear_lang=UNSET):
+    def process_post(self, language_session_key=UNSET, accept=UNSET, preview_lang=UNSET, clear_lang=UNSET):
         """
-        Build a request and then process it using the ``DarkLangMiddleware``.
+        Build the POST request and set the language settings according to parameters
 
         Args:
             language_session_key (str): The language code to set in request.session[LANUGAGE_SESSION_KEY]
             accept (str): The accept header to set in request.META['HTTP_ACCEPT_LANGUAGE']
-            preview_lang (str): The value to set in request.GET['preview_lang']
-            clear_lang (str): The value to set in request.GET['clear_lang']
+            preview_lang (str): The value to set in request.POST['preview_lang']
+            clear_lang (str): The value to set in request.POST['clear_lang']
         """
         session = {}
         set_if_set(session, LANGUAGE_SESSION_KEY, language_session_key)
 
         meta = {}
         set_if_set(meta, 'HTTP_ACCEPT_LANGUAGE', accept)
 
-        get = {}
-        set_if_set(get, 'preview-lang', preview_lang)
-        set_if_set(get, 'clear-lang', clear_lang)
+        post = {}
+        set_if_set(post, 'preview_lang', preview_lang)
+
+        if clear_lang is True:
+            set_if_set(post, 'reset', 'reset')
+        else:
+            set_if_set(post, 'set_language', 'set_language')
+
+        request = Mock(
+            spec=HttpRequest,
+            session=session,
+            META=meta,
+            POST=post,
+            user=UserFactory()
+        )
+
+        # Process the incoming request to set the Language
+        Darklang().process_darklang_request(request)
```

Comment:
As I mentioned, might as well just make process_darklang_request a method outside of an object as Darklang has no member variables.